src/algebra/homology/homotopy.lean

```diff
@@ -0,0 +1,507 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.homology.additive
+
+/-!
+# Chain homotopies
+
+We define chain homotopies, and prove that homotopic chain maps induce the same map on homology.
+-/
+
+universes v u
+
+open_locale classical
+noncomputable theory
+
+open category_theory category_theory.limits homological_complex
+
+variables {ι : Type*}
+variables {V : Type u} [category.{v} V] [has_zero_object V] [preadditive V]
+
+variables {c : complex_shape ι} {C D E : homological_complex V c}
+variables (f g : C ⟶ D) (h k : D ⟶ E) (i : ι)
+
+section
+
+/-- Auxiliary definition for `homotopy`. Use `homotopy.from_next` instead. -/
+def from_next' (f : Π i j, C.X i ⟶ D.X j) (i j : ι) : C.X_next i ⟶ D.X j :=
+match c.next i with
+| none := 0
+| some ⟨i',w⟩ := (C.X_next_iso w).hom ≫ f i' j
+end
+
+lemma from_next'_eq (f : Π i j, C.X i ⟶ D.X j) {i i' : ι} (w : c.rel i i') (j : ι) :
+  from_next' f i j = (C.X_next_iso w).hom ≫ f i' j :=
+begin
+  dsimp [from_next'],
+  rw c.next_eq_some w,
+  refl,
+end
+
+@[simp] lemma from_next'_zero (i j : ι) : from_next' (λ i j, (0 : C.X i ⟶ D.X j)) i j = 0 :=
+begin
+  dsimp [from_next'],
+  rcases c.next i with ⟨⟩|⟨⟨i', w⟩⟩;
+  { dsimp [from_next'._match_1], simp, },
```

Comment:
```suggestion
  { dsimp [from_next'], simp, },
```
... and similarly in a few other proofs in this file.

Reply:
Done, thanks.